File: Java/BookyardClient/src/practice/bookyard/client/LoginDialogEventLoop.java
package practice.bookyard.client;

import java.awt.Dimension;
import java.awt.Toolkit;

import javax.swing.JDialog;

public class LoginDialogEventLoop implements Runnable {

	@Override
	public void run() {
		
		// Create and set title
		JDialog dialog = new LoginDialog("Bookyard Login");
		
		// Set default close behavior
		dialog.setDefaultCloseOperation(JDialog.DISPOSE_ON_CLOSE);
		
		// Set size
		dialog.setSize(400, 300);
		// dialog.setResizable(false);
		
		// Set position on the screen
		Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
		dialog.setLocation((int)screenSize.getWidth() / 2, (int)screenSize.getHeight() / 2);
		dialog.setLocationRelativeTo(null);
		
		dialog.setVisible(true);
	}

}

File: Client/src/LoginDialogEventLoop.kt
package bookyard.client;

import java.awt.Dimension;
import java.awt.Toolkit;

import javax.swing.JDialog;

public class LoginDialogEventLoop : Runnable {

    override public fun run() {

        // Create and set title
        val dialog : JDialog = LoginDialog("Bookyard Login");

        // Set default close behavior
        dialog.setDefaultCloseOperation(JDialog.DISPOSE_ON_CLOSE);

        // Set size
        dialog.setSize(400, 300);
        // dialog.setResizable(false);

        // Set position on the screen
        val screenSize : Dimension = Toolkit.getDefaultToolkit().getScreenSize();
        dialog.setLocation(screenSize.getWidth().toInt() / 2, screenSize.getHeight().toInt() / 2);
        dialog.setLocationRelativeTo(null);

        dialog.setVisible(true);
    }
}